### DISCUSSION TITLE
File processing time?

### DISCUSSION THREAD

**Discussion by SpinozaWeb** [DrewThomasson/ebook2audiobook#19] [Q&A] [answered]
Thanks for the new docker image. Installed - works) I have a laptop with intel i7-10875h and rtk 2070 max-q.
I converted the text in Russian. But even after 2.5 hours the conversion was not completed.
File size 800 kb
How long does the process take for you?

**Comment by DrewThomasson** [comment, marked as answer]
For me? 

- Keep in mind the only way to get major speed ups is by having a CUDA capable Nvidia graphics card :/

## The example book used for these test cases was the first book of guardians of ga'hoole
- which is a 5 hour audiobook

## With a 4GB VRAM NVIDA(forgot the name)
- around real time reading speed 
- around 5-6 hours 

## With a 12 gb VRAM Nvidia (3060?)
-A bit faster than real time reading speed
-around 4-3 hours 

## With only cpu on desktop (i9)?
- around slower than real time reading speed
- around 7-8 hours

## With only CPU on laptop?
- holy crap it's so slow 
- like 5 days

**Reply by lexmirnov** [reply]
How to enable GPU speedup for a non-Docker install on Windows? I have a 3080Ti, but the app won't use it.
![image](https://github.com/user-attachments/assets/fc2783e8-aefe-4703-9e13-51ae889c3e2c)

_Edit; nvm, didn't have CUDA installed._

**Reply by DrewThomasson** [reply]
> How to enable GPU speedup for a non-Docker install on Windows? I have a 3080Ti, but the app won't use it.
> 
> ![image](https://github.com/user-attachments/assets/fc2783e8-aefe-4703-9e13-51ae889c3e2c)
> 
> 
> 
> _Edit; nvm, didn't have CUDA installed._
> 
> 

Good to hear u got it working in your end!
 nice 👍

**Reply by freeshard** [reply]
Hello and thanks for the topic and for the project, is there any special scripts or settings which need to be tuned for enabling Cuda? I have old video card but with cuda asus turbo 1070 8gb, what else should i do except just install CUDA Toolkit from Nvidia site?

**Reply by DrewThomasson** [reply]
I think?

Unsure I'm usually lazy and just use the docker.

Then CUDA works without a headache

🤷

(I'll look into this tho as windows uses an automated install script now for v2.0)

**Reply by shadeau9** [reply]
I was trying to get my GPU (RTX3090) to be used with this on my Windows machine. I installed CUDA toolkit 12.6 based on @lexmirnov comment, but that still didn't use the GPU. After a few hours of digging it turned out that the version of torch installed by default with pip to meet the Coqui XTTSv2 requirements isn't the version that has cuda enabled. 

To fix the issue I ran the cmd file to set up the environment, exited out of the program, activated the new conda environment, and ran `pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu124` which installed the cuda-enabled version of torch. Once it finished installing I ran the cmd again, selected the GPU and it's been running for a while without issues. 

Hope this helps some others having problems.

**Reply by ROBERT-MCDOWELL** [reply]
@shadeau9 with python 3.12?

**Reply by ROBERT-MCDOWELL** [reply]
in any case for now xttsv2 seems to accept only cuda 11.8 max. but if 12.4 works for you so well done.

**Reply by shadeau9** [reply]
@ROBERT-MCDOWELL I confirmed that the python environment created by the cmd is Python 3.12 and it did generate chapter wav files using my GPU after I installed `torch 2.5.1+cu124`. I'm having issues merging them into a completed file, but I think that's more related to the book title having a "?" in it. If there's any outputs or things you'd like to see from my environment I'm happy to share.

**Reply by ROBERT-MCDOWELL** [reply]
having a title with ? char should n't be a problem.
what's the title of the book?

**Reply by ROBERT-MCDOWELL** [reply]
we really should find a way to help the user to install the right cuda since torch, xtts-v2 and python itself can install different default version which can be wrong.

**Reply by shadeau9** [reply]
> having a title with ? char should n't be a problem. what's the title of the book?

Good to know, thank you! Looking into it further there was also a missing chapter wav file, which may have contributed. Regardless, I'm trying it again with some different parameters to see if it works. I don't want to derail this thread though so I'll try a few things on my end to see if I can figure it out.

Thanks for the quick responses!

**Reply by ROBERT-MCDOWELL** [reply]
no worries. I suggest to wait for the next git update to make your test. something like dozen of bugs/mistake/typo have been fixed...

**Reply by ricardocavalcante** [reply]
Why not Brazilian Portuguese? The population of Brazil is 10 times the Portugal...

**Reply by DrewThomasson** [reply]
> Why not Brazilian Portuguese? The population of Brazil is 10 times the Portugal...

It looks like XTTS (The default model for ebook2audiobook) supports Brazilian Portuguese.  If you select `Portuguese` and then give it a sample of someone speaking Brazilian Portuguese as the voice to clone then it should be able to say it in a Brazilian Portuguese accent.

ref - > https://github.com/coqui-ai/TTS/discussions/4127#discussioncomment-11800681

ALSO:

Please further discuss this in a different chat/issue or open a new one, as I don't think your question relates to this discussion at all...

**Reply by j2shortt** [reply]
5 days..........Ouch... Well, I'm converting my first book now....... it took 500 seconds-ish to reach 0.1%..... Based on the math its looking like just under 6 days to complete...... ooooof...... we shall see. lol...

**Comment by DrewThomasson** [comment]
If your really looking for fast on CPU tho, I have a repo that uses piper-tts instead (cause piper is also multi-lingual)

- But it's based on piper-tts (so no voice cloning you can only select from the pre-existing list)

-Unless you train your own model lol

- And it's also Siri quality voices


https://github.com/DrewThomasson/ebook2audiobookpiper-tts

- piper tts audio output file demo ⬇️

[piper-tts-demo](https://github.com/user-attachments/assets/7d2328b9-ac65-4485-b1b3-fe1006f041c6)

- piper-tts huggingface demo space ⬇️

https://huggingface.co/spaces/drewThomasson/ebook2audiobookpiper-tts

**Comment by SyamsQ** [comment]
How to run GPU? I have rtx 3060
![Screenshot 2025-06-01 010331](https://github.com/user-attachments/assets/63a97f5f-380d-4fc1-af6e-cd713d1b9c1a)
![Screenshot 2025-06-01 010352](https://github.com/user-attachments/assets/59518f84-fe07-4cdd-8cee-77bcde615111)

**Reply by shadeau9** [reply]
I wrote a guide for how to set this up with an Nvidia GPU in another discussion thread. I'd suggest following [those instructions](https://github.com/DrewThomasson/ebook2audiobook/discussions/477) and commenting in that discussion thread.

**Reply by SyamsQ** [reply]
I have already installed Ebook2audiobook. Do I need to reinstall it after installing the Cuda Toolkit?

**Reply by suskozaver** [reply]
no, just run:

to check if u sucesfully installed the toolkit: `docker run --rm -it --gpus=all nvcr.io/nvidia/k8s/cuda-sample:nbody nbody -gpu -benchmark `

u should see something similar to this at the bottom:
> Compute 8.6 CUDA device: [NVIDIA GeForce RTX 3080]
69632 bodies, total time for 10 iterations: 56.565 ms
= 857.180 billion interactions per second
= 17143.598 single-precision GFLOP/s at 20 flops per interaction

if you didn't get such results, perhaps just run: `docker run --rm --gpus all nvidia/cuda:11.0-base nvidia-smi`

after the tools were installed for sure, just run ebook2audiobook with: 

`docker run --pull always --rm --gpus all -p 7860:7860 athomasson2/ebook2audiobook`

**Comment by PublicAB** [comment]
If the text is too long, an error will occur. Can this be resolved


Analyzing numbers, maths signs, dates and time to convert in words...
filter_chapter() error: expected string or buffer
Traceback (most recent call last):
  File "/app/lib/functions.py", line 780, in filter_chapter
    text = clock2words(text, lang, lang_iso1, tts_engine, is_num2words_compat)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/functions.py", line 1227, in clock2words
    return time_rx.sub(repl_num, text)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/functions.py", line 1217, in repl_num
    phrase = lc["past"].format(hour=n2w(h), minute=n2w(mnt)) if mnt != 0 else lc["oclock"].format(hour=n2w(h))
                                    ^^^^^^
  File "/app/lib/functions.py", line 1173, in n2w
    word = math2words(n, lang, lang_iso1, tts_engine, is_num2words_compat)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/functions.py", line 1252, in math2words
    text = re.sub(r'(\d)\)', r'\1 : ', text)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/regex/regex.py", line 278, in sub
    return pat.sub(repl, string, count, pos, endpos, concurrent, timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: expected string or buffer
Caught DependencyError: filter_chapter() error: expected string or buffer
get_chapters() failed!

**Reply by ROBERT-MCDOWELL** [reply]
this has nothing to do with this discussion. open an issue, and provide the whole log from your command line until the except. provide also the file you want to convert if possible with the settings you used (which tts etc..)
